pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/Functions.java

```diff
@@ -893,4 +893,42 @@ void putFunctionState(String tenant, String namespace, String function, Function
      */
     CompletableFuture<Void> putFunctionStateAsync(
             String tenant, String namespace, String function, FunctionState state);
+
+    /**
+     * Sends update function request to worker leader. This is an internal only api
+     * <p/>
+     *
+     * @param tenant
+     *            Tenant name
+     * @param namespace
+     *            Namespace name
+     * @param function
+     *            Function name
+     * @param functionMetaData
+     *            byte repr of FunctionMetaData
+     **
+     * @throws NotAuthorizedException
+     *             You don't have admin permission to get the configuration of the cluster
+     * @throws PulsarAdminException
+     *             Unexpected error
+     */
+    void updateOnWorkerLeader(String tenant, String namespace, String function, byte[] functionMetaData,
```

Comment:
I don't think it is a good idea to expose this in the admin API. This is an internal thing and users should not have to the option to call this directly

Reply:
removed